hamilton/function_modifiers/macros.py

```diff
@@ -773,65 +810,234 @@ def final_result(upstream_int: int) -> int:
                 return upstream_int
 
         In all likelihood, you should not be using this, and this is only here in case you want to expose a node for
-        consumption/output later. Setting the namespace in individual nodes as well as in ``pipe`` is not yet supported.
+        consumption/output later. Setting the namespace in individual nodes as well as in ``pipe_input`` is not yet supported.
+
+    3. ``on_input`` -- this selects which input we will run the pipeline on.
+        In case ``on_input`` is set to None (default), we apply ``pipe_input`` on the first parameter. Let us know if you wish to expand to other use-cases.
+
+        The following would apply function *_add_one* and *_add_two* to ``p2``:
+
+        .. code-block:: python
+
+            @pipe_input(
+                step(_add_one)
+                step(_add_two, y=source("upstream_node")),
+                on_input = "p2"
+            )
+            def final_result(p1: int, p2: int, p3: int) -> int:
+                return upstream_int
+
+        .. |
+            For extra control in case of multiple function arguments (parameters), we can also specify the target parameter that we wish to transform.
+            In case ``on_input`` is set to None (default), we apply ``pipe_input`` on the first parameter only. If ``on_input`` is set for a specific transform
+            make sure the other ones are also set either through a global setting or individually, otherwise it is unclear which transforms target which parameters.
+
+            The following applies *_add_one* to ``p1``, ``p3`` and *_add_two* to ``p2``
+
+            .. code-block:: python
+
+                @pipe_input(
+                    step(_add_one).on_input(["p1","p3"])
```

Comment:
Update docs -- we disabled this (for now), right?

Reply:
Correct, I found a hack how to "comment out" things from `rst` so Sphinx doesn't generate this part of the docstring.